src/generic.jl

```diff
@@ -30,11 +30,14 @@ Writes an array of Avro objects if there is a
 write(Encoder, typeof(ArraySchema.items), T) method.
 """
 function write(encoder::Encoder, schema::ArraySchema, value::Vector{T}) where T
-    bytes_written = encode_long(encoder, Int64(length(value)))
+    l = Int64(length(value))
+    bytes_written = encode_long(encoder, l)
     for item in value
         bytes_written += write(encoder, schema.items, item)
     end
-    bytes_written += encode_byte(encoder, zero(UInt8))
+    if l != 0
+        bytes_written += encode_byte(encoder, zero(UInt8))
```

Comment:
What is this zero for in general and why is 0 length special cased?

Reply:
I've added comments and a link to the spec.

Reply:
Thanks.